dot/core/service.go

```diff
@@ -512,6 +525,10 @@ func (s *Service) createBlockResponse(blockRequest *network.BlockRequestMessage)
 
 	switch c := blockRequest.StartingBlock.Value().(type) {
 	case uint64:
+		if c == 0 {
```

Comment:
I am curious to know why when someone does a blockrequest with starting block 0 we silently change it to 1 and move on ?

Reply:
block 0 is the genesis block, so it has no parent, so it will error if we attempt to add it to the block tree. it's handled on the receiver side but also better not to send it in the first place, since we already know that the peer has the same genesis block as us